src/annoylib.h

```diff
@@ -833,14 +991,15 @@ template<typename S, typename T, typename Distance, typename Random>
   }
 
   void _get_all_nns(const T* v, size_t n, size_t search_k, vector<S>* result, vector<T>* distances) {
-    Node* v_node = (Node *)malloc(_s); // TODO: avoid
-    memcpy(v_node->v, v, sizeof(T)*_f);
+    Node* v_node = (Node *)calloc(_s, 1); // TODO: avoid
+    memcpy(v_node->v, v, sizeof(T) * _f);
     D::init_node(v_node, _f);
 
     std::priority_queue<pair<T, S> > q;
 
-    if (search_k == (size_t)-1)
-      search_k = n * _roots.size(); // slightly arbitrary default value
+    if (search_k == (size_t)-1) {
+      search_k = D::default_search_k(n, _roots.size());
```

Comment:
why would different distance functions have different implementations? I would rather make it consistent